### ISSUE TITLE
[Feature requested] ListView do not ScrollTo a group when there is no child of this group.

### ISSUE THREAD

**Issue by jessiezh0320** [xamarin/Xamarin.Forms#8279]
### Description

At present, we must special a item object  as the first parameter of method : ScrollTo .
When a group of listview has no child, the function `ScrollTo ` cannot scroll to the corresponding group. 

Many customers think it is not reasonable, and it should have this feature. 

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

**Comment by techduggu** [comment]
I'd like to pick this up. Currently trying to reproduce it and I also noticed that it doesn't Scroll if list item (first parameter) is passed as null (in-case I'd like to scroll to a group only). For e.g.

`List.ScrollTo(null, mygroupitem, ScrollToPosition.MakeVisible, true);`

@jsuarezruiz @jfversluis  Is it expected behavior or should I fix this too?

Thanks!

**Comment by jfversluis** [comment]
Makes sense to me to scroll to the group whenever the item is not found but the group is. I guess the cases for the different parameter values would be:

- No item, do have a group: go to the group
- Do have an item, no group: find the first item throughout all the groups and scroll there
- Have an item and have a group: find the item in the specified group and scroll there. If the item isn't found, scroll to the group
- No item, no group: doesn't make any sense, do nothing

Right?

**Comment by Rajeswari06** [comment]
We are facing the same issue in iPad-iOS 12.4.1. with grouped listView.

We have used a grouped listview, when we try to expand and collapse an item in bottom of the screen scroll position is changing to top and unable to see the selected item. 

Listview.scroll is not working in grouped listview if it don't have items.

**Comment by jfversluis** [comment]
There is a PR for the Android side of things @Rajeswari06 if you're brave enough you could add the iOS implementation? 😉

**Comment by techduggu** [comment]
@Rajeswari06 When I fixed for Android, I had seen the existing implementation for iOS too as it also uses the `TemplatedItemsList.cs` from Core library and it requires little tweaks at the iOS `ListViewRenderer.cs` at the following LOC:

```
if (results.Item1 == -1 || (results.Item1 == -1 && results.Item2 == -1))
return;
```
You can refer the PR #8310 for more details. I couldn't complete the iOS as I don't have mac/iOS but if you'd like to pick this up, I'm happy to help should you have any further questions.

**Comment by samhouts** [comment]
closed by #8310
